### PR TITLE
Client-node rewritten to ESModules

### PR DESCRIPTION
**Pull request by lotas**
Having CJS only was blocking some of the dependency upgrades which switched to ESM only distributions

`stream-buffers` was removed since it was only used in tests and could be rewritten using Readable/Writable streams from stdlib

### CODE REVIEW

File: clients/client/src/upload.js [hunk 1]
```diff
@@ -62,7 +62,7 @@ const upload = async ({
   const retryCfg = {
     retries: retries === undefined ? 5 : retries,
     delayFactor: delayFactor === undefined ? 100 : delayFactor,
-    randomizationFactor: randomizationFactor === undefined ? randomizationFactor : 0.25,
+    randomizationFactor: randomizationFactor === undefined ? 0.25 : randomizationFactor,
```
Comment (lotas) [inline clients/client/src/upload.js:65]: for some reason it didn't cause any problem before node20 upgrade

Comment (matt-boris) [reply clients/client/src/upload.js:65]: 😬

Comment (matt-boris) [reply clients/client/src/upload.js:65]: Nice catch!

**Review by matt-boris — APPROVED** [review]
Great thanks Yarik 🎉

**Pull request merged**